algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -165,15 +165,11 @@ void sort_by_key_via_sort(
       Kokkos::RangePolicy<ExecutionSpace>(exec, 0, n),
       KOKKOS_LAMBDA(int i) { permute(i) = i; });
 
-// FIXME OPENMPTARGET The sort happens on the host so we have to copy keys there
-#ifdef KOKKOS_ENABLE_OPENMPTARGET
   auto keys_in_comparator = Kokkos::create_mirror_view(
-      Kokkos::view_alloc(Kokkos::HostSpace{}, Kokkos::WithoutInitializing),
+      Kokkos::view_alloc(Kokkos::HostSpace{}, Kokkos::WithoutInitializing,
+                         exec),
```

Comment:
`exec` can't access `HostSpace`.
```suggestion
      Kokkos::view_alloc(Kokkos::HostSpace{}, Kokkos::WithoutInitializing),
```

Reply:
@masterleinad should passing `exec` be fully removed, or only in guarded cases e.g. with sycl?

Reply:
fwiw guarding against sycl for the create_mirror_view got me back to compilation, though the test still seg faults as expected

```diff
@@ -165,11 +165,15 @@ void sort_by_key_via_sort(
       Kokkos::RangePolicy<ExecutionSpace>(exec, 0, n),
       KOKKOS_LAMBDA(int i) { permute(i) = i; });
 
+#ifndef KOKKOS_ENABLE_SYCL
   auto keys_in_comparator = Kokkos::create_mirror_view(
       Kokkos::view_alloc(Kokkos::HostSpace{}, Kokkos::WithoutInitializing,
                          exec),
       keys);
   Kokkos::deep_copy(exec, keys_in_comparator, keys);
+#else
+  auto keys_in_comparator = keys;
+#endif
 
   static_assert(sizeof...(MaybeComparator) <= 1);
   if constexpr (sizeof...(MaybeComparator) == 0) {
```

Reply:
@masterleinad your suggestion resolved the test failure with sycl, thanks! May we commit it or is that a sycl-specific change and requires some guarding?

Reply:
Thanks for the explanation, very helpful 👍